src/components/table/column-picker/column-picker.component.ts

```diff
@@ -176,11 +171,31 @@ export class ColumnPickerComponent implements OnChanges {
     }
 
     /** A function that can be called to add columns. If no columns are passed to the function, the items that are selected in the left column will be added. */
-    addColumns(columns: ReadonlyArray<string | ColumnPickerGroupItem> = this._deselectedSelection): void {
-        columns = columns.filter(column => this.selected.indexOf(this.getColumnName(column)) === -1);
+    addColumns(columns: ReadonlyArray<string | ColumnPickerGroupItem | ColumnPickerTreeNode> = this._deselectedSelection): void {
```

Comment:
What is the situation where `ColumnPickerTreeNode`s can be mixed in here? I've checked where it is called and the type indicates `ReadonlyArray<string | ColumnPickerGroupItem>` in every case.